### PR TITLE
Upgrade node to 18

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
Fixes: #2525. This PR updates Node.js to the current LTS: `18.x.x`. 

## What was done
* Upgrade to Node.js in version `18.x.x` in:
	* package.json
	* nvmrc
	* docker
* Update GHA scripts to point to `.nvmrc` files instead of the hardcoded version in workflows
* Split `build` and `start` npm scripts with generating types into `prebuild` & `prestart` (for better readability)

### Broken webpack

Starting from Node.js version `17.x.x` maintainers deprecate legacy OpenSSL [provider](https://github.com/nodejs/node/issues/40455). This is causing our Webpack to throw `Error: error:0308010C:digital envelope routines::unsupported` while running any command. Suggested [workaround](https://github.com/webpack/webpack/issues/14532) is to add `NODE_OPTIONS=--openssl-legacy-provider` every time we use Webpack in our scripts. There is no fix in the current Webpack release yet 😞.

What we can do?
1) Stay on version `16.x.x` for now and upgrade the Webpack to the newest version (in other PR)
2) Stay on version `18.x.x` and add flags plus upgrade the Webpack to the newest version (in other PR). In this solution, we can't use Node.js in `16.x.x` as `NODE_OPTIONS` are incompatible between 16 and 18.

---

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [x] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants

CONTAINERS=2

### CODE REVIEW

**Comment by lkostrowski** [comment]
cc @zaiste xmas is earlier 🎁 ;)

File: .npmrc [hunk 1]
```diff
@@ -1 +1,2 @@
-puppeteer_skip_chromium_download = true
+puppeteer_skip_chromium_download=true
+legacy-peer-deps=true
```
Comment (orzechdev) [inline .npmrc:2]: If no in this PR, we should still update peer-depts one day

Comment (krzysztofzuraw) [reply .npmrc:2]: Sure - I want to keep changes to a minimum and then introduce proper deps in next PR

**Comment by zaiste** [comment]
> cc @zaiste xmas is earlier 🎁 ;)

This can be one of the best presents I ever received ;)

**Review by tomaszszymanski129 — APPROVED** [review]

File: .npmrc (hunk 1 above)
Comment (NyanKiyoshi) [reply .npmrc:2]: Can you create a task for this so it's doesn't get forgotten?

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-14.19.0
+18.12.0
```
Comment (NyanKiyoshi) [inline .nvmrc:1]: Can you not simply put v18 so it uses the latest version instead of a fixed version?

**Review by andrzejewsky — APPROVED** [review]
So much gratefulness within one PR <3 see @krzysztofzuraw you made them a day :D

Let's test this carefully. The node is just a builder for us so if the building process works we can merge it

File: .npmrc (hunk 1 above)
Comment (krzysztofzuraw) [reply .npmrc:2]: https://github.com/saleor/saleor-dashboard/issues/2549

**Review by NyanKiyoshi — APPROVED** [review]

**Review by orzechdev — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -266,18 +266,20 @@
...
     "check-types": "tsc --noEmit",
     "extract-json-messages": "formatjs extract 'src/**/*.{ts,tsx}' --out-file locale/defaultMessages.json --format scripts/formatter.js",
     "extract-messages": "npm run extract-json-messages",
     "fetch-schema": "graphql-codegen --config ./fetch-schema.yml && node scripts/build-schema.js",
     "heroku-postbuild": "npm run build",
     "serve:lhci": "NODE_ENV=production npm run server",
-    "start": "npm run build-types && webpack-dev-server -d",
-    "storybook": "start-storybook -p 3000 -c src/storybook/",
+    "prestart": "npm run build-types",
+    "start": "env NODE_OPTIONS=--openssl-legacy-provider webpack-dev-server -d",
```
Comment (andrzejewsky) [inline package.json:281]: is that webpack thing? Please create an issue for that and link it in this PR

**Review by andrzejewsky — DISMISSED** [review]

**Review by szczecha — DISMISSED** [review]

**Review by NyanKiyoshi — DISMISSED** [review]

File: package.json (hunk 3 above)
Comment (krzysztofzuraw) [reply package.json:281]: Done in https://github.com/saleor/saleor-dashboard/issues/2557

**Review by szczecha — APPROVED** [review]

**Review by orzechdev — APPROVED** [review]

**Review by karolm-saleor — APPROVED** [review]
lgtm

**Review by andrzejewsky — APPROVED** [review]
to the moon 🚀

**Pull request merged**